### PR TITLE
Update directives to remove netcoreapp3.1

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #1031 

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/Shared/ServiceProviderExtensions.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@
 // limitations under the License.
 // </copyright>
 
-#if NETFRAMEWORK || NETSTANDARD2_0_OR_GREATER || NETCOREAPP3_1_OR_GREATER
+#if NETFRAMEWORK || NETSTANDARD2_0_OR_GREATER || NET6_0_OR_GREATER
```
[the remark this answers is not in the document]
Comment (vishweshbankwar) [reply src/Shared/ServiceProviderExtensions.cs]: Good catch removed

**Review by xiang17 — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

File: src/Shared/ServiceProviderExtensions.cs
Comment (utpilla) [inline src/Shared/ServiceProviderExtensions.cs:1]: This file is not used by any component. It can be removed.

**Review by Kielek — APPROVED** [review]

File: src/Shared/ServiceProviderExtensions.cs
Comment (vishweshbankwar) [reply src/Shared/ServiceProviderExtensions.cs:1]: done

**Comment by twenzel** [comment]
👍🏻

**Pull request merged**